app/helpers/ResponseDecorator/IResponseDecorator.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Helpers;
+
+/**
+ * Interface for decorators, which augment the presenter output just before it is passed on to JSON serialization.
+ */
+interface IResponseDecorator
```

Comment:
Do I understand correctly that there will always be only one decorator?

Reply:
Yes. However, If multiple decorators were needed, I will implement ComposedDecorator, which could have multiple decorators under it.

Reply:
Okay, the only thing that seems a little bit unclear is what is a good place for `$payload->jsonSerialize()` to happen. What if the decorator received an already serialized payload?

Reply:
Good point. I thought it is straightforward, but the composition will be a bit more difficult. I will give it some thought.